### PR TITLE
Use autosize

### PR DESCRIPTION
**Pull request by dorechen**
Use autosize with TextArea. It helps the text area grow in height as more lines of text are inputted

### CODE REVIEW

File: src/TextArea.js [hunk 1]
```diff
@@ -24,6 +26,18 @@ const baseTextArea = css`
...
   }
 `
-export const TextArea = props => (
-  <textarea css={[baseTextArea]} data-enable-grammarly="false" {...props} />
-)
+export const TextArea = props => {
+  const textareaRef = useRef(null)
+  useEffect(() => {
+    autosize(textareaRef.current)
+  })
```
Comment (gouegd) [inline src/TextArea.js]: you can limit this to happen only when the `props.value` changes by extracting it and specifying it like this:
`useEffect(someFunction, [value])`

Comment (gouegd) [reply src/TextArea.js]: (dont forget to pass `value` below if it's extracted from props)

File: src/TextArea.js [hunk 2]
```diff
@@ -26,11 +27,14 @@ const baseTextArea = css`
...
+export const TextArea = ({ value, ...props }) => {
   const textareaRef = useRef(null)
-  useEffect(() => {
-    autosize(textareaRef.current)
-  })
+  useEffect(
+    () => {
+      autosize(textareaRef.current)
+    },
+    { value }
```
Comment (gouegd) [inline src/TextArea.js]: this should be an array, not an object: `[ value ]`

and you forgot to pass `value` to the <textarea> below

Comment (dorechen) [reply src/TextArea.js]: can you please explain to me why it's important to do these? (or what I should read up on)

Comment (dorechen) [reply src/TextArea.js]: sorry for the amateur question 😅

Comment (gouegd) [reply src/TextArea.js]: This avoids unnecessary calls to autosize, as we know it's only ever useful to call it when the value has changed.
See https://reactjs.org/docs/hooks-reference.html#conditionally-firing-an-effect

**Pull request merged**